public/app/features/variables/datasource/adapter.ts

```diff
@@ -25,7 +25,7 @@ export const createDataSourceVariableAdapter = (): VariableAdapter<DataSourceVar
       }
       return false;
     },
-    setValue: async (variable, option, emitChanges = false) => {
+    setValue: async (variable, option, emitChanges = true) => {
```

Comment:
This change makes sure that all panels are refreshed when we change data source variables.